mne/preprocessing/artifact_detection.py

```diff
@@ -333,3 +334,189 @@ def _annotations_from_mask(times, art_mask, art_name):
             durations.append(times[l_idx[-1]] - times[l_idx[0]])
         desc.append(art_name)
     return Annotations(onsets, durations, desc)
+
+
+@verbose
+def annotate_break(raw, events=None,
+                   min_duration=15.,
+                   t_start_after_previous=5.,
+                   t_stop_before_next=5.,
+                   ignore=('bad', 'edge'),
+                   *,
+                   verbose=None):
+    """Create `~mne.Annotations` for breaks in an ongoing recording.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The continuous data to analyze.
+    events : None | array, shape (n_events, 3)
+        If ``None`` (default), operate based solely on the annotations present
+        in ``raw``. If an events array, ignore any annotations in the raw data,
+        and operate based on these events only.
+    min_duration : float
+        The minimum time span in seconds between the offset of one and the
+        onset of the subsequent annotation (if ``events`` is ``None``) or
+        between two consecutive events (if ``events`` is an array) to consider
+        this period a "break". Defaults to 15 seconds.
+    t_start_after_previous, t_stop_before_next : float
+        Specifies how far the to-be-created "break" annotation extends towards
+        the two annotations or events spanning the break. This can be used to
+        ensure e.g. that the break annotation doesn't start and end immediately
+        with a stimulation event. If, for example, your data contains a break
+        of 30 seconds between two stimuli, and ``t_start_after_previous`` is
+        set to ``5`` and ``t_stop_before_next`` is set to ``3``, the break
+        annotation will start 5 seconds after the first stimulus, and end 3
+        seconds before the second stimulus, yielding an annotated break of
+        ``30 - 5 - 3 = 22`` seconds. Both default to 5 seconds.
+
+        .. note:: The beginning and the end of the recording will be annotated
+                  as breaks, too, if the period from recording start until the
+                  first annotation or event (or from last annotation or event
+                  until recording end) is at least ``min_duration`` seconds
+                  long.
+
+    ignore : iterable of str | None
+        Annotation descriptions starting with these strings will be ignored by
+        the break-finding algorithm. The string comparison is case-insensitive,
+        i.e., ``('bad',)`` and ``('BAD',)`` are equivalent. By default, all
+        annotation descriptions starting with "bad" and annotations
+        indicating "edges" (produced by data concatenation) will be
+        ignored. If ``None``, take all existing annotations into account.
```

Comment:
I would not allow None, they can just pass `()` or `[]`

Reply:
Addressed in 70423eb